modules/exploits/linux/misc/hp_jetdirect_path_traversal.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require "rex/proto/pjl"
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::SNMPClient
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'HP Jetdirect Path Traversal Arbitrary Code Execution',
+      'Description'    => %q{
+        The module exploits a path traversal via Jetdirect to gain arbitrary code execution by
+        writing a shell script that is loaded on startup to /etc/profile.d. Then, the printer
+        is restarted using SNMP. Impacted printers:
+        HP PageWide Managed MFP P57750dw
+        HP PageWide Managed P55250dw
+        HP PageWide Pro MFP 577z
+        HP PageWide Pro 552dw
+        HP PageWide Pro MFP 577dw
+        HP PageWide Pro MFP 477dw
+        HP PageWide Pro 452dw
+        HP PageWide Pro MFP 477dn
+        HP PageWide Pro 452dn
+        HP PageWide MFP 377dw
+        HP PageWide 352dw
+        HP OfficeJet Pro 8730 All-in-One Printer
+        HP OfficeJet Pro 8740 All-in-One Printer
+        HP OfficeJet Pro 8210 Printer
+        HP OfficeJet Pro 8216 Printer
+        HP OfficeJet Pro 8218 Printer
+      },
+      'Author'         => [
+        'Jacob Baines', # Python PoC
+        'Matthew Kienow <matthew_kienow[AT]rapid7.com>', # Metasploit module
```

Comment:
Ah, next time remember to add your e-mail to https://github.com/rapid7/metasploit-framework/blob/master/.mailmap, that way you won't need to manually put your e-mail here all the time :-)

Reply:
Awesome!